tests/nnstreamer_datarepo/datareposink/unittest_datareposink.cc

```diff
@@ -0,0 +1,540 @@
+#include <gtest/gtest.h>
+#include <glib.h>
+#include <glib/gstdio.h>
+#include <gst/gst.h>
+#include <unittest_util.h>
+
+/**
+ * @brief Bus callback function
+ */
+static gboolean
+bus_callback (GstBus * bus, GstMessage * message, gpointer data)
+{
+  switch (GST_MESSAGE_TYPE (message)) {
+    case GST_MESSAGE_EOS:
+      g_main_loop_quit ((GMainLoop *) data);
+      break;
+    default:
+      break;
+  }
+
+  return TRUE;
+}
+
+/**
+ * @brief Test for writing image files
+ */
+TEST (datareposink, writeImageFiles)
+{
+  GError *error = NULL;
+  GFile *file = NULL;
+  gchar *contents = NULL;
+  gchar *filename = NULL;
+  GstBus *bus;
+  GMainLoop *loop;
+  gint i = 0;
+
+  loop = g_main_loop_new (NULL, FALSE);
+
+  gchar *str_pipeline =
+      g_strdup ("gst-launch-1.0 videotestsrc num-buffers=10 ! pngenc ! "
+      "datareposink location=png_%02d.png json=image.json");
+
+  GstElement *pipeline = gst_parse_launch (str_pipeline, NULL);
+  g_free (str_pipeline);
+  ASSERT_NE (pipeline, nullptr);
+
+  bus = gst_pipeline_get_bus (GST_PIPELINE (pipeline));
+  gst_bus_add_watch (bus, bus_callback, loop);
+  gst_object_unref (bus);
+
+  gst_element_set_state (pipeline, GST_STATE_PLAYING);
+  g_main_loop_run (loop);
+
+  gst_element_set_state (pipeline, GST_STATE_NULL);
```

Comment:
Can we guarantee that all files are saved by datareposink when EOS occurs?
How about checking whether 10 files are arrived or not?

Reply:
Thank you.